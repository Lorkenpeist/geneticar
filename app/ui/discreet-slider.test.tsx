import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import DiscreetSlider from "./discreet-slider";

describe(DiscreetSlider.name, () => {
  const slider = (
    <DiscreetSlider
      id="propId"
      name="propName"
      label="My Property"
      min={1}
      max={5}
      step={1}
      defaultValue={3}
    />
  );

  it("renders correctly", () => {
    const tree = render(slider);
    expect(tree.container).toMatchSnapshot();
  });

  it("updates its value when dragged", async () => {
    const tree = render(slider, { container: document.createElement("form") });

    const sliderInput = tree.getByRole("slider");

    fireEvent.change(sliderInput, { target: { value: 4 } });

    // Confirm that the display text is in sync with the slider
    expect(tree.getByText(/My Property:/)).toHaveTextContent(
      /^My Property: 4$/,
    );
    // Confirm that the form will be submitted with the correct value
    expect(tree.container).toHaveFormValues({ propName: "4" });
  });
});
