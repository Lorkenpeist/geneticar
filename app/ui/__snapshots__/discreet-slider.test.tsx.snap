// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscreetSlider renders correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center text-xs font-medium"
  >
    <label
      for="propId"
    >
      My Property
      : 
      3
    </label>
    <div
      class="flex flex-row gap-1 items-center"
    >
      <div
        class="w-5 text-right"
      >
        1
      </div>
      <input
        id="propId"
        list="propId:markers"
        max="5"
        min="1"
        name="propName"
        step="1"
        type="range"
        value="3"
      />
      <datalist
        id="propId:markers"
      >
        <option
          value="1"
        />
        <option
          value="2"
        />
        <option
          value="3"
        />
        <option
          value="4"
        />
        <option
          value="5"
        />
      </datalist>
      <div
        class="w-7 text-left"
      >
        5
      </div>
    </div>
  </div>
</div>
`;
